src/neuroconv/utils/json_schema.py

```diff
@@ -306,3 +307,14 @@ def get_metadata_schema_for_icephys():
     )
 
     return schema
+
+
+def validate_metadata(metadata: Dict[str, dict], schema: Dict[str, dict], verbose: bool = False):
+    """Validate metadata against a schema."""
+    encoder = NWBMetaDataEncoder()
+    # The encoder produces a serialiazed object so we de serialized it for comparison
+    serialized_metadata = encoder.encode(metadata)
+    decoded_metadata = json.loads(serialized_metadata)
```

Comment:
Just out of curiosity, can you not just specify the class as one of the inputs to `json.loads`?